memcache/memcache.go

```diff
@@ -485,6 +568,196 @@ func (c *Client) GetMulti(keys []string) (map[string]*Item, error) {
 	return m, err
 }
 
+//MetaGet function returns value for a cached item given a key. It also accepts few MetaGetFlags
+//which supports additional functionality.
+//ErrCacheMiss is returned if a key is not found
+//The key must be at most 250 bytes in length.
+func (c *Client) MetaGet(key string, metaGetFlags *MetaGetFlags) (metaGetResponse *MetaGetResponse, err error) {
+	err = c.withKeyAddr(key, func(addr net.Addr) error {
+		return c.metaGetFromAddr(addr,
+			key,
+			metaGetFlags,
+			func(metaGetRes *MetaGetResponse) { metaGetResponse = metaGetRes })
+	})
+	return
+}
+
+func (c *Client) metaGetFromAddr(addr net.Addr, key string, metaGetFlags *MetaGetFlags, cb func(response *MetaGetResponse)) error {
+
+	return c.withAddrRw(addr, func(rw *bufio.ReadWriter) error {
+		metaGetCommandFlags := createMetaGetFlagCommands(metaGetFlags)
+		if _, err := fmt.Fprintf(rw, "mg %s %s\r\n", key, metaGetCommandFlags); err != nil {
+			return err
+		}
+		if err := rw.Flush(); err != nil {
+			return err
+		}
+
+		if err := parseMetaGetResponse(rw.Reader, cb); err != nil {
+			return err
+		}
+		return nil
+
+	})
+
+}
+
+//function parses meta get response
+func parseMetaGetResponse(r *bufio.Reader, cb func(*MetaGetResponse)) error {
+
+	response, err := r.ReadSlice('\n')
+	if err != nil {
+		return err
+	}
+
+	//check for CLIENT_ERROR or SERVER_ERROR
+	if err = getMemCacheErrorFromResponse(response); err != nil {
+		return err
+	}
+
+	//Check if cache miss has occurred
+	if bytes.HasPrefix(response, metaGetMissPrefix) {
+		return ErrCacheMiss
+	}
+
+	responseComponents := strings.Fields(string(response))
+
+	//metaGet command response will be in the format
+	//VA <size> <flags>*\r\n
+	//<data block>\r\n
+	//Check if at least VA and <size> is present in the response
+	if len(responseComponents) < 2 || !bytes.HasPrefix(response, metaGetValuePrefix) {
+		return fmt.Errorf("memcache: unexpected line in meta get response: %q", response)
+	}
+
+	var size int
+	//read value size
+	if size, err = strconv.Atoi(responseComponents[1]); err != nil {
+		return err
+	}
+
+	metaGetResponse := new(MetaGetResponse)
+	metaGetResponse.Value = make([]byte, size+2)
+
+	//populate value
+	_, err = io.ReadFull(r, metaGetResponse.Value)
+
+	if err != nil {
+		metaGetResponse.Value = nil
+		return err
+	}
+
+	//check if value has a suffix of clrf (i.e. \r\n)
+	if !bytes.HasSuffix(metaGetResponse.Value, crlf) {
+		metaGetResponse.Value = nil
+		return fmt.Errorf("memcache: corrupt meta get result read")
+	}
+
+	var metaRespMetadata *MetaResponseMetadata
+	metaGetResponse.Value = metaGetResponse.Value[:size]
+
+	responseMetadata := responseComponents[2:]
```

Comment:
Would we not want to store the two character response value for all meta commands in the MetaResponseMetadata object?

Reply:
You mean "VA" for meta get command ? This is constant and theres no point in storing those. I am returning the tokens in the response which the user has selected.